pkg/mount/raw_mount.go

```diff
@@ -0,0 +1,170 @@
+			allowedDirs:   allowedDirs,
+			kl:            keylock.New(),
+			trashLocation: trashLocation,
+		},
+	}
+	if err := rm.Load(rootSubstrings); err != nil {
+		return nil, err
+	}
+	return rm, nil
+}
+
+func (rm *rawMounter) Reload(rootSubstring string) error {
+	newRBM, err := NewRawBindMounter(
+		[]string{rootSubstring},
+		rm.mountImpl,
+		rm.allowedDirs,
+		rm.trashLocation,
+	)
+	if err != nil {
+		return err
+	}
+	return rm.reload(rootSubstring, newRBM.mounts[rootSubstring])
+}
+
+func shouldSkipMountPoint(mountPoint string) bool {
+	for _, skip := range skippedLocations {
+		if strings.HasPrefix(mountPoint, skip) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// this mount filtering implementation is done based on logic implemented in findmnt + libmount
+func (rm *rawMounter) Load(rawVolumeDevicesPaths []string) error {
+	mountPoints, err := GetMounts()
+	if err != nil {
+		return err
+	}
+
+	// try to find all bind mounts of raw volumes
+	if len(rawVolumeDevicesPaths) == 0 || rawVolumeDevicesPaths[0] == "" {
+		mountPointsByMajMin := make(map[string]*[]mount.Info)
+		mountPointsByTarget := make(map[string]*mount.Info)
+
+		for _, mp := range mountPoints {
+			// skip proc mount points
+			if shouldSkipMountPoint(mp.Mountpoint) || mp.Root == "/null" {
+				continue
+			}
+			majMin := fmt.Sprintf("%v:%v", mp.Major, mp.Minor)
+
+			mountPointsForNumber, exists := mountPointsByMajMin[majMin]
+			if !exists {
+				mountPointsForNumber = &[]mount.Info{}
+				mountPointsByMajMin[majMin] = mountPointsForNumber
+			}
+
+			*mountPointsForNumber = append(*mountPointsForNumber, *mp)
+
+			mountPointsByTarget[mp.Mountpoint] = mp
+		}
+
+		devicesMP, exists := mountPointsByTarget["/dev"]
+		if !exists {
+			return fmt.Errorf("cannot find /dev mount point while loading mount points")
+		}
+
+		devMajMin := fmt.Sprintf("%v:%v", devicesMP.Major, devicesMP.Minor)
+
+		mountPointsForNumber := mountPointsByMajMin[devMajMin]
+
+		mps := *mountPointsForNumber
+		filteredMPs := []mount.Info{}
+		for i := range mps {
+			if mps[i].Mountpoint != "/dev" {
+				filteredMPs = append(filteredMPs, mps[i])
+			}
+		}
+
+		// TODO: when raw volume is mounted a few times on the same node, that case is not handled
+		// skipping dev mount point
+		for _, mp := range filteredMPs {
+			devicePath := "/dev" + mp.Root
+
+			// source for raw volumes is equal to rawVolumeDevicePath
+			mount, ok := rm.mounts[devicePath]
+			if !ok {
+				mount = &Info{
+					Device: devicePath,
+					Fs:     "",
+					Minor:  mp.Minor,
+					Mountpoint: []*PathInfo{&PathInfo{
+						Root: normalizeMountPath(devicePath),
+						Path: normalizeMountPath(mp.Mountpoint),
+					}},
+				}
+				rm.mounts[devicePath] = mount
+			}
+		}
+	}
+
+	// find raw volume bind mounts
+	for _, rawVolumeDevicePath := range rawVolumeDevicesPaths {
+		var mountPointForRoot *mount.Info
+		for _, mp := range mountPoints {
+			if strings.HasSuffix(rawVolumeDevicePath, mp.Root) {
```

Comment:
The rawVolumeDevicePath should be a suffix like `/dev/pxd/pxd`. So this should change to
```
strings.HasPrefix(mp.Root, rawVolumeDevicePath)
```
Again this depends on what is mp.Root set to. I believe it is set to the whole device path